### PR TITLE
[DEV-1984] Create schools_updater.yml

### PR DESCRIPTION
**Pull request by yodarjun**
This script does the following:
1.Fetch a JSON object from the schools API endpoint using credentials.
2.Convert the JSON response into a CSV file.
3.Commit and push the updated CSV file back to the repository.

### CODE REVIEW

File: .github/workflows/schools_updater.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          in2csv data.json > schools.csv
+
+      - name: Commit and push CSV file
+        run: |
+          # Configure git
+          git config --local user.name "github-actions[bot]"
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+
+          # Commit and push changes
+          git add path/to/your/output.csv
```
Comment (erinosher) [inline .github/workflows/schools_updater.yml:70]:
```suggestion
          git add schools.csv
```

File: .github/workflows/schools_updater.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            # Increment page number for the next request
+            PAGE=$((PAGE + 1))
+          done
+
+      - name: Convert JSON to CSV
+        run: |
+          # Convert JSON to CSV and save as output.csv
+          in2csv data.json > schools.csv
+
+      - name: Commit and push CSV file
```
Comment (erinosher) [inline .github/workflows/schools_updater.yml:63]:
```suggestion
      - name: Commit CSV file
```

File: .github/workflows/schools_updater.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+name: Fetch Paginated JSON and Update CSV
+
+on:
+  schedule:
+    - cron: "0 0 * * *"  # Runs once a day at midnight (UTC)
+  workflow_dispatch:    # Allows for manual runs as well
+  push:
+    branches:
+      - aa_schools_csv_updater
+
```
Comment (erinosher) [inline .github/workflows/schools_updater.yml:10]: Remove before merging.

```suggestion

```

**Review by erinosher — CHANGES_REQUESTED** [review]
Updated to run on your dev branch. Looks fine other than the commit/push step isn't working. Made some suggestions that _should_ fix it.

File: .github/workflows/schools_updater.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            jq -s '.[0] + .[1]' $AGGREGATE_FILE <(echo "$PAGE_DATA") > tmp.json && mv tmp.json $AGGREGATE_FILE
+
+            # Increment page number for the next request
+            PAGE=$((PAGE + 1))
+          done
+
+      - name: Convert JSON to CSV
+        run: |
+          # Convert JSON to CSV and save as school.csv
+          in2csv data.json > schools.csv
```
Comment (erinosher) [inline .github/workflows/schools_updater.yml:61]: We only actually care about the name column, particularly given the size of the CSV.

File: .github/workflows/schools_updater.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        run: |
+          # Initialize page and limit parameters
+          PAGE=1
+          LIMIT=250
+          AGGREGATE_FILE=data.json
+          echo "[]" > $AGGREGATE_FILE  # Initialize an empty JSON array
+
+          while true; do
+            # Fetch a page of data from the API
+            RESPONSE=$(curl -s -H "Authorization: Bearer $API_TOKEN" "$API_URL?page=$PAGE&limit=$LIMIT")
```
Comment (erinosher) [inline .github/workflows/schools_updater.yml:43]: needs to use the `verified[eq]=true` filter, I tried updating it, but it errored for me for some reason.

File: .github/workflows/schools_updater.yml (hunk 3 above)
Comment (erinosher) [reply .github/workflows/schools_updater.yml:10]: Leaving this as aa_schools_csv_updater for now as we want to ensure this is working before it hits main

**Review by erinosher — CHANGES_REQUESTED** [review]

File: .github/workflows/schools_updater.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+      - name: Commit CSV file
+        run: |
+          # Configure git
+          git config --local user.name "github-actions[bot]"
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+
+          # Commit and push changes
+          git add schools.csv
+          git commit -m "Update CSV file from API data [skip ci]"
```
[the remark this answers is not in the document]
Comment (yodarjun) [reply .github/workflows/schools_updater.yml]: the `skip ci` basically tells github actions that it doesnt have to run any checks for this commit

[the remark this answers is not in the document]
Comment (erinosher) [reply .github/workflows/schools_updater.yml]: Oh, that shouldn't matter here b/c the commit is on the `GITHUB_TOKEN` so it [won't trigger anyway.](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow)

File: .github/workflows/schools_updater.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+          mv schools.csv path/to/your/schools.csv
+
+          # Configure git
+          git config --local user.name "github-actions[bot]"
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+
+          # Commit and push changes
+          git add path/to/your/school.csv
+          git commit -m "Update school CSV file from API data [skip ci]"
+          git push
```
Comment (erinosher) [inline .github/workflows/schools_updater.yml:88]:
```suggestion
        run: |
          # Configure git
          git config --local user.name "github-actions[bot]"
          git config --local user.email "github-actions[bot]@users.noreply.github.com"

          # Commit and push changes
          git add schools.csv
          git commit -m "Update schools CSV file from API data [skip ci]"
          git push
```

File: .github/workflows/schools_updater.yml [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+            jq -s '.[0] + .[1]' $AGGREGATE_FILE <(echo "$PAGE_DATA") > tmp.json && mv tmp.json $AGGREGATE_FILE
+
+            # Increment page number for the next request
+            PAGE=$((PAGE + 1))
+          done
+
+      - name: Filter JSON to only include school names and save as school.csv
+        run: |
+          # Extract only the "name" field from each entry and overwrite data.json
+          jq -r '.[].name' data.json > schools.csv
```
Comment (erinosher) [inline .github/workflows/schools_updater.yml:74]:
```suggestion
        run: |
          # Add first line to CSV file
          echo "\"Below is a list of all schools that we've manually verified. If you see one that hasn't been included, please add it via my.mlh.io and we'll verify it soon.\"" > schools.csv
          
          # Extract only the "name" field from each entry and overwrite data.json
          jq -r '.[].name' data.json >> schools.csv
```

**Review by erinosher — APPROVED** [review]

File: .github/workflows/schools_updater.yml [hunk 9]
```diff
@@ -0,0 +1,88 @@
+name: Fetch Paginated JSON and Update CSV
+
+on:
+  schedule:
+    - cron: "0 0 * * *"  # Runs once a day at midnight (UTC)
+  workflow_dispatch:    # Allows for manual runs as well
+  push:
+    branches:
+      - main
+      - aa_schools_csv_updater
```
Comment (erinosher) [inline .github/workflows/schools_updater.yml:10]:
```suggestion
      - main
```

**Pull request merged**